web-client/src/pages/admin/AdminSockDetailsPage.tsx

```diff
@@ -0,0 +1,244 @@
+import React, { useState, useEffect } from "react";
+import { useParams } from "react-router-dom";
+import toast from "react-hot-toast";
+import {
+  useGetSockById,
+  useUpdateSockMutation,
+} from "@/api/inventory/queries";
+import { Button } from "@/components/ui/button";
+import { Table } from "@/components/ui/table";
+import EditItemModal from "@/components/EditItemModal";
+import EditVariantModal from "@/components/EditVariantModal";
+import AddSizeModal from "@/components/AddSizeModal";
+import { Edit, Plus } from "lucide-react";
+import { SockVariant, UpdateSock } from "@/api/inventory/model";
+import { NO_IMAGE_PLACEHOLDER } from "@/shared/constants";
+
+export default function AdminSockDetailsPage() {
+  const { sockId } = useParams<{ sockId: string }>();
+  const numericSockId = Number(sockId);
+
+  const { data: sock, refetch, isLoading, isError } = useGetSockById(numericSockId);
+
+  const [isEditSockOpen, setEditSockOpen] = useState(false);
+  const [editVariant, setEditVariant] = useState<null | SockVariant>(null);
+  const [isAddVariantOpen, setAddVariantOpen] = useState(false);
+  const [currentVariants, setCurrentVariants] = useState<SockVariant[]>([]);
+
+  const updateSockMutation = useUpdateSockMutation();
+
+  useEffect(() => {
+    if (sock?.variants) {
+      setCurrentVariants(sock.variants);
+    }
+  }, [sock]);
+
+  const handleUpdateSock = async (updatedSock: UpdateSock) => {
+    try {
+      const payload = {
+        id: numericSockId,
+        ...updatedSock,
+      };
+      await updateSockMutation.mutateAsync(payload);
+      refetch();
+      setEditSockOpen(false);
+      toast.success("Sock updated successfully!");
+    } catch (error) {
+      console.error("Error updating sock:", error);
+      toast.error("Failed to update sock details.");
+    }
+  };
+
+  const handleAddVariant = async (newVariant: SockVariant) => {
+    try {
```

Comment:
Just for info: typically you don't want to wrap "everything" within the "try" block. Only the piece that can actually throw an error (in this case I believe the call to the mutation).

It's usually best to have independent try-catch blocks for each individual error. But Javascript is weird with errors :/